vscode/server.ts

```diff
@@ -59,6 +59,7 @@ async function applySettings() {
 }
 
 async function compileAndValidate() {
+  let hasCompilationError = false;
```

Comment:
This is a bit confusingly named, because the errors below are also compilation errors. A better name would probably be `let compilationFailed = false`

Reply:
Thanks @Ekrekr, I agree. I have made the suggested change and synced this branch with dataform repo's main branch.